pyinstrument/magic/magic.py

```diff
@@ -241,6 +243,13 @@ def pyinstrument(self, line, cell=None):
             )
             return
 
+        # If a KeyboardInterrupt occurred during the magic execution,
+        # send a SIGINT signal to prevent further executions.
+        if isinstance(cell_result.error_in_exec, KeyboardInterrupt):
+            # The traceback is already shown during the cell execution above, so we
+            # don't re-raise the exception.
+            sys.exit(SIGINT)
```

Comment:
I found a way to have no additional traceback nor warnings shown. This requires temporarily registering a custom exception handler and raising a custom exception:

```suggestion
            old_custom_tb = ip.CustomTB
            old_custom_exceptions = ip.custom_exceptions

            def _silent_exception_handler(self, etype, value, tb, tb_offset=None):
                """No-op handler for silent exception."""
                # restore the original exception handler
                ip.CustomTB = old_custom_tb
                ip.custom_exceptions = old_custom_exceptions

            ip.set_custom_exc((InterruptSilently, ), _silent_exception_handler)

            raise InterruptSilently()
```

where:

```python

class InterruptSilently(Exception):
    """Exception used to interrupt execution without showing traceback"""
```

This results in much cleaner:

![image](https://github.com/user-attachments/assets/7efaf4d9-c7a3-493c-ba8d-dfa4731c2bce)